ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -640,6 +749,16 @@ public static short getVersion(String idStr) throws IOException {
           accountId = stream.readShort();
           containerId = stream.readShort();
           break;
+        case BLOB_ID_V5:
+          blobIdFlag = stream.readByte();
+          type = BlobIdType.values()[blobIdFlag & BLOB_ID_TYPE_MASK];
+          isEncrypted = (blobIdFlag & IS_ENCRYPTED_MASK) != 0;
+          int dataTypeOrdinal = (blobIdFlag & BLOB_DATA_TYPE_MASK) >> BLOB_DATA_TYPE_SHIFT;
+          blobDataType = dataTypeOrdinal < BlobDataType.values().length ? BlobDataType.values()[dataTypeOrdinal] : null;
```

Comment:
I don't think we should set it to `null` here based on the check. If the ser-deser etc went right, there should _never_ be a reason for `dataTypeOrdinal >= BlobDataType.values().length`. I'd rather see it fail if that happens (with the `IndexOutOBoundsException` that the enum will throw)

Reply:
ok